src/main/scala/org/ergoplatform/explorer/persistence/dao/HeaderDao.scala

```diff
@@ -0,0 +1,57 @@
+package org.ergoplatform.explorer.persistence.dao
+
+import cats.implicits._
+import doobie.implicits._
+import doobie.refined.implicits._
+import doobie.{ConnectionIO, Update}
+import org.ergoplatform.explorer._
+import org.ergoplatform.explorer.persistence.models.Header
+
+object HeaderDao extends Dao {
+
+  val tableName: String = "node_headers"
+
+  val fields: List[String] = List(
+    "id",
+    "parent_id",
+    "version",
+    "height",
+    "n_bits",
+    "difficulty",
+    "timestamp",
+    "state_root",
+    "ad_proofs_root",
+    "transactions_root",
+    "extension_hash",
+    "miner_pk",
+    "w",
+    "n",
+    "d",
+    "votes",
+    "main_chain"
+  )
+
+  def update(h: Header): ConnectionIO[Header] =
+    update
+      .withUniqueGeneratedKeys[Header](fields: _*)(h -> h.id)
+
+  def updateMany(list: List[Header]): ConnectionIO[List[Header]] =
+    update
+      .updateManyWithGeneratedKeys[Header](fields: _*)(list.map(h => h -> h.id))
+      .compile
+      .to[List]
+
+  def get(id: Id): ConnectionIO[Option[Header]] =
+    (selectFr ++ fr"where id = $id").query[Header].option
```

Comment:
Doesn't look like a good way to do DB queries. Any better approach?
Should this boilerplate be manually written for each table?
If yes, better to find an alternative

Reply:
There is no production-ready alternative to doobie in Typelevel stack.

Reply:
In the book Doobie is used in a different way, and I suggest to follow that pattern.
We need to keep SQL statements explicit and don't compose them from small fragments.
I understand the idea to have generic generator of SQL, but having fixed data model, which is not supposed to be changed it looks like an overkill and it makes understanding and troubleshooting much more complex.

Reply:
Yeah, that sounds reasonable. Let’s use fragments for table name and fields only in order to avoid listing them manually. It should not harm readability.